File: tests/bindings/vbnet/TestBrickletIndustrialDualRelay.vb
Imports Tinkerforge

Module TestBrickletIndustrialDualRelay
    Const HOST As String = "localhost"
    Const PORT As Integer = 4223
    Const UID As String = "xyz" ' Change to your UID

    Sub Main()
        Dim ipcon As New IPConnection() ' Create IP connection
        Dim dr As New BrickletIndustrialDualRelay(UID, ipcon) ' Create device object

        ipcon.Connect(HOST, PORT) ' Connect to brickd
        ' Don't use device before ipcon is connected

        ' Turn both relays off and on
        dr.SetValue(False, False)
        System.Threading.Thread.Sleep(1000)
        dr.SetValue(True, True)

        ipcon.Disconnect()
    End Sub
End Module

File: tests/bindings/octave/test_bricklet_industrial_dual_relay.m
function test_bricklet_industrial_dual_relay()
    more off;
    
    HOST = "localhost";
    PORT = 4223;
    UID = "xyz"; % Change to your UID

    ipcon = java_new("com.tinkerforge.IPConnection"); % Create IP connection
    dr = java_new("com.tinkerforge.BrickletIndustrialDualRelay", UID, ipcon); % Create device object

    ipcon.connect(HOST, PORT); % Connect to brickd
    % Don"t use device before ipcon is connected

    % Turn both relays off and on
    dr.setValue(false, false);
    pause(1);
    dr.setValue(true, true);

    ipcon.disconnect();
end

test_bricklet_industrial_dual_relay();
